datefinder.py

```diff
@@ -196,7 +206,12 @@ def parse_date_string(self, date_string, captures):
         if len(date_string) < 3:
             return None
 
-        as_dt = parser.parse(date_string)
+        try:
```

Comment:
I think it makes sense to let dateutil bubble up the exception here. Point is, datefinder shouldn't be passing anything to dateutil we don't already believe it will parse.